### PR TITLE
feat(nav-pilot): install --frozen for CI og onboarding (#572)

### PR DESCRIPTION
**Pull request by Starefossen**
`install` honorerer allerede den pinnede revisjonen i `.nav-pilot/agentpakke.lock.json` (#597). Det som gjensto for CI og onboarding var å **nekte å prompte** og **nekte å bevege pinnen**. Dette er den flata. Lukker den siste `--frozen`-kula i `Begrensninger i dag`, og hører til #572.

## Hva den nekter, og hvorfor

Ledespørsmålet har vært: hva vil en CI-jobb heller feile på enn å pakke inn?

- **Ingen erklæring.** Da er det ingenting å fryse mot.
- **Erklæring uten `sha`.** En upinnet install henter det default-grenen tilfeldigvis var den morgenen — to kjøringer et døgn fra hverandre installerer ulikt innhold. Det er nettopp forskjellen pinnen finnes for.
- **`sha` som ikke er førti tegn.** git nekter en forkortet objekt-id i en fetch. Uten flagget kommer dette som en misvisende git-feil (#607); `--frozen` sier hva som faktisk er galt.
- **En annen revisjon enn den erklærte.** Alle veier dit er stengt av flaggsjekkene, og sammenligningen står der som setningen som holder det slik.
- **En delvis install** — konflikt-hoppede filer, elementer målscopet ikke kan holde, eller navn manifestet fører opp som kilden ikke har. Uten `--frozen` er alle tre en advarsel, og pinnen skrives etterpå som om alt gikk inn. En CI-jobb som installerer ufullstendig og melder suksess er nøyaktig feilen flagget finnes for, så dette er den ene dokumenterte begrensninga `--frozen` **lukker** framfor å arve. Nektinga ligger før JSON-utskriften, så `--json` heller ikke kan melde en halv install grønn — og `--json` rapporterer nå `skipped` ved siden av `installed` og `conflicts`, som den ikke gjorde.
- **En enkeltartefakt-install** (`install <artefakt>`): den skriver ingen erklæring og følger ikke `items`, så det er ingenting å holde den til.
- **En Tier 2-agentpakke.** `--frozen` dekker dem ikke. En Tier 2-install pinner én revisjon per *bruker* og digest-verifiserer hvert payload-tre mot manifestet sitt — den er allerede frossen i konstruksjonen, og `guardPakkeScope` nekter alle andre scope enn brukerens, så det finnes ingen repo-scopet install for erklæringa å holde den til. Meldinga ber deg installere den én gang *uten* `--frozen`.

Erklæringa skrives aldri under `--frozen` — ikke engang tilbake som den verdien den allerede har. En CI-jobb skal ikke lage diff i en fil den bare var ment å adlyde.

## Exit-koder

| Kode | Betydning |
| --- | --- |
| `0` | Alt gikk inn, og det som ligger der er den pinnede revisjonen |
| `1` | Installasjonen feilet: kilden utilgjengelig, motstridende flagg, ikke-kontraktsmessig pakke |
| `3` | Installasjonen feilet ikke, men pinnen ble ikke honorert: ingen erklæring, ingen brukbar pinne, en annen revisjon enn den erklærte, én enkelt artefakt eller en Tier 2-pakke framfor det erklæringa navngir, eller en delvis install |

`3` er med vilje ikke `0`, samme holdning som `scripts/nav-pilot-golden.sh` har til sin egen `3`: noe som ikke ble bevist, er ikke grønt. Det er `3` mot `1` som lar en jobb skille «pinnen er ikke det repoet sier» fra «installasjonen røk».

## Samspill med de andre flaggene

- **Motsier og avvises som bruksfeil (exit 1):** `--source`, `--ref`, `--type`, `--user`. Bruker-scope leser aldri en erklæring, så det finnes ingen pinne å holde der — og det svarer `1` fra begge dører nå, ikke `1` fra flaggsjekken og `3` fra `frozenPrecheck` for den samme feilen.
- `--dry-run --frozen` nekter med `3` når arbeidstreet allerede avviker fra den pinnede revisjonen: konfliktsjekken kjører før tørrkjøringa. En tørrkjøring som svarte «ville installert» over et tre den ekte installasjonen nekter, er verre enn ingen tørrkjøring.
- Et `items`-navn pakka ikke har, stopper hele installasjonen med `1`, ikke `3`: det er en erklæring som ikke stemmer med pakka, ikke en pinne som ikke ble honorert.
- **Forenlige:** `--force` (beveger ikke pinnen, gjør bare treet likt den — og er veien gjennom konflikt-nektinga), `--dry-run`, `--json`.
- **Prompter aldri:** scope-spørsmålet hoppes over, og `install --frozen` uten navn ber om navnet framfor å åpne en velger.

## `sync` får ikke flagget

`sync --apply` finnes for å flytte pinnen, og `sync` uten `--apply` skriver ingenting fra før — et `--frozen` der ville vært enten et synonym eller en selvmotsigelse. «Er pinnen fersk?» svarer `nav-pilot sync` allerede på med exit `1`.

## Hva gjennomgangen fant, og hva som ble gjort

Tre ting sto igjen etter første runde. Alle tre er nå rettet, og hver rettelse har en test som er kjørt mot koden uten den:

1. **`frozenComplete` dekket to av tre halve installasjoner.** Den sjekket `Conflicts` og `Unsupported`, men ikke navn manifestet fører opp som kilden ikke har. `installArtifact` skriver `⚠ Agent not found: ghost` og går videre, så en `collections/test/manifest.json` med `["a","ghost"]` installerte 1 av 2 agenter og svarte exit 0 — på default-kilden, som har `collections/` og ikke noe agentpakke-manifest, altså navn som ikke er utledet av resolveren. `--json` skrev et suksessdokument uten `skipped`-nøkkel i det hele tatt. `Skipped int` er nå `Missing []string`, så nektinga kan si hvilke navn som manglet, og `skipped` er med i JSON-en.
2. **`--frozen` kunne aldri installere en Tier 2-pakke, og nektinga pekte feil vei.** `frozenPrecheck` krever repo-scope; `payloadOnly` rutet til `guardPakkeScope`, som nekter alt annet enn bruker-scope og ba deg kjøre `--user` — et flagg `--frozen` selv avviser. **Valget ble å nekte med riktig begrunnelse framfor å få det til å virke:** en Tier 2-install er allerede pinnet og digest-verifisert per revisjon, så «frossen» er redundant der framfor manglende. Å få det til å virke ville betydd enten en repo-scopet pinne ingen launch leser, eller at `--frozen` godtar `--user` og dermed slutter å handle om erklæringsfila. Det ville også krevd at `recordDeclaration` skriver for Tier 2, som den ikke gjør i dag (den ligger etter `payloadOnly`-returen), slik at nektingas råd om «kjør én gang og commit fila» hadde vært sant.
3. **`--type`-kontrollen beviste ikke det som ble påstått.** Påstanden var at flagget faktisk installerte uten nektinga. Det stemmer om koden, men `TestFrozenContradictoryFlags/--type_agent` feilet av en tilfeldig grunn — katalogen var ikke et git-repo — ikke fordi en installasjon landet. Testen kjører nå i et repo med erklæring og en kilde som lar seg slå opp, og sjekken finnes også i `cmdInstallAuto`, slik at flaggsjekken er en beleilighet framfor det eneste som står mellom `--frozen` og en installasjon uten precheck, uten pinne-sammenligning og uten fullstendighetssjekk.

Kontrollkjøringene, hver med den nye vakta fjernet:

```
# 1: manifestet navnga «ghost», kilden har den ikke
--- FAIL: TestFrozenRefusesMissingArtifact
    a frozen install landed 1 of 2 agents and reported success
--- FAIL: TestSkippedIsVisibleInJSON/a_plain_install_reports_it
    --json does not report what it skipped: {"installed": 1, "conflicts": 0, "unsupported": null, ...}

# 2: Tier 2
--- FAIL: TestFrozenRefusesTier2
    exited 1, want ExitFrozen (3): ... Install it for your user: nav-pilot install --user grillmester
    the refusal is still guardPakkeScope's scope complaint

# 3: --type, med flaggsjekken fjernet — den dype sjekken tar den (exit 3, ikke 1)
--- FAIL: TestFrozenContradictoryFlags/install_test-a_--frozen_--type_agent
    exited 3, want ExitError (1)
# og med begge fjernet — installasjonen lander:
  ✓ test-a
  ✓ Added agent "test-a".
--- FAIL: TestFrozenContradictoryFlags/install_test-a_--frozen_--type_agent
    the contradiction was accepted

# bruker-scope med to exit-koder
--- FAIL: TestFrozenRefusesUserScope
    user scope exited 3 here and 1 from run(): one mistake, two answers
```

## Tester

Tretten tester, alle kjørt mot koden med hver enkelt vakt slått av: hver vakt har minst én test som faller uten den, og av riktig grunn — `--type`-tilfellet feilet før på at katalogen ikke var et git-repo, og feiler nå på at en installasjon lander.

`TestFrozenInstallsPinnedRevisionWithoutTouchingTheFile`, `TestFrozenRefusesPartialInstallAndForceGetsThrough`, `TestFrozenRefusesUnsupportedKinds` og `TestFrozenRefusesSingleArtifact` går gjennom **ekte git**: `gitAgentpakke` bygger et lokalt repo med to commits, `RemoteURLFn` peker `navikt/grillmester` dit, og installasjonen kjører den virkelige resolveren. `resolveSource` stubbes bare der poenget er at den *aldri kalles*, eller at en revisjon som ikke er den erklærte, blir avvist.

`mise run nav-pilot:check` er grønn.

### CODE REVIEW

**Comment by Starefossen** [comment]
Tredje runde på denne. Gjennomgangen sa merge, men fant én ting til som er verdt å ta først, og den er nå rettet i `9e…`.

**Den fjerde halve installasjonen**

En pakke med *både* layout og payloads er verken Tier 2 for `frozenTier2` eller ufullstendig for `frozenComplete`. `payloadOnly` krever at `layout` mangler, så en blandet pakke tok Tier 1-veien: installerte layout-halvdelen, staget ingen payload, og returnerte grønt. Oppstarten dør senere i `mixedPakkeRefusal` — lenge etter at CI har blitt grønn.

Det er nøyaktig familien av feil denne PR-en finnes for å lukke, og påstanden «tre måter en install lander ufullstendig» var altså feil med én.

Første forsøk på rettelsen traff ikke: jeg utvidet `frozenTier2` til å dekke hver payload-bærende pakke, men kallet sto fortsatt på `payloadOnly`-grenen, som en blandet pakke aldri tar. Testen fanget det. Kallet er flyttet opp foran tier-delingen. `TestFrozenRefusesMixedPakke` feiler uten rettelsen med «a frozen install of a mixed pakke was accepted: the layout landed and no payload was staged».

**Begrunnelsen var også for sterk**

Nektelsen hvilte på at en payload-install «allerede er frossen i konstruksjonen». Det er den ikke, i den betydningen dette flagget har: brukerscope leser ingen erklæring (`scopeDeclaration` returnerer nil der), så en install der løser standardgrenen, og to kjøringer et døgn fra hverandre kan pinne hver sin revisjon — nøyaktig feilen `--frozen` finnes for å hindre. Digest-verifisering beviser payload-treets integritet, ikke hvilken revisjon som ble hentet.

Nektelsen står, men på riktig grunnlag: et payload-tre pinnes per bruker, ikke per repo, så det finnes ingen repo-scopet install for erklæringa å holde det til. Å pinne en payload per repo er reelt arbeid og en egen beslutning under #572. Både kommentaren og dokumentasjonen sier nå det.

**Fra gjennomgangen ellers, verifisert og rent**

`Skipped int` → `Missing []string` har én produsent og ingen tidligere ikke-test-leser, så ingen aritmetikk ble brutt og nil-mot-tom er ikke skillbart noe sted. `len(Missing) > 0` har ingen legitim «fraværende med vilje»-sak: en `items`-innsnevring fjerner navn før `installItems`, en ikke-støttet type går til `Unsupported`, og for en agentpakke-kilde kommer manifestet fra resolveren selv, så et spøkelse er strukturelt umulig. `Missing` er bare nåbar der navnene er håndskrevne — altså en `collections/*/manifest.json`-kilde, som er standardkilden.

Begge `--type`-kontrollene ble reprodusert: flagglag-vakten fjernet gir exit 3 der 1 var ventet, begge fjernet lander en faktisk installasjon.

**Én rettelse i briefen min:** ingen vakt ble fjernet fra `installAllFromSource` — den hadde aldri noen. Utilgjengeligheten stemmer likevel, sporet gjennom `cmdInstallAll` som bare nås under `--user`.

File: cli/nav-pilot/internal/cli/main_test.go [hunk 1]
```diff
@@ -514,8 +514,8 @@ func TestInstallAgent_NotFound(t *testing.T) {
 	if err != nil {
 		t.Fatal(err)
 	}
-	if result.Skipped != 1 {
-		t.Errorf("skipped = %d, want 1", result.Skipped)
+	if len(result.Missing) != 1 {
+		t.Errorf("skipped = %d, want 1", len(result.Missing))
```
[the remark this answers is not in the document]
Comment (Starefossen) [reply cli/nav-pilot/internal/cli/main_test.go]: Fixed — the three assertions now say `missing`, matching the field. Worth keeping distinct: `skipped` is the JSON key for the count, `Missing` is the struct field that names them.

File: docs/README.agentpakke.md [hunk 2]
```diff
@@ -347,6 +347,32 @@ Erklæringen ligger over config-nøkkelen fordi den er repoets gjennomgåtte val
 - `nav-pilot sync --apply` flytter `sha`. Da kommer oppdateringa som én linje diff i en pull request framfor som usynlig lokal state. En sync uten `--apply` skriver aldri: den skal ikke skitne til en fil noen må committe. Sync bumper en revisjon, men peker aldri repoet på en annen agentpakke — det er en `install`.
 - `nav-pilot sync` sier fra når kilden scopet er registrert mot, er en annen enn den repoet erklærer. Scopet vinner (valget er per scope), og pinnen står urørt — men da ville de to ellers drevet fra hverandre i stillhet, så det skrives én linje om det, med begge veiene ut.
 
+**`--frozen` er CI-varianten.** `nav-pilot install <navn> --frozen` installerer nøyaktig det repoet erklærer, eller lar være. Den spør aldri om noe, og den flytter aldri pinnen — ikke engang for å skrive den tilbake som den verdien den allerede har, for en CI-jobb skal ikke lage diff i en fil den bare var ment å adlyde.
+
+Den nekter, før noe hentes eller skrives:
```
[the remark this answers is not in the document]
Comment (Starefossen) [reply docs/README.agentpakke.md]: Korrekt, og rettet. Tier-sjekkene trenger manifestet, som ligger i kilden, så de kan ikke komme før kilden er resolvet.

Teksten sier nå hvilken garanti som gjelder hvor: alltid før noe skrives, og for de fire første punktene også før noe hentes. Det er skrivingen som er den garantien som betyr noe for et scope, men da må teksten si akkurat det og ikke mer.

File: cli/nav-pilot/internal/cli/install.go [hunk 3]
```diff
@@ -180,6 +183,14 @@ func finishInstall(err error, flagSource string, dryRun, scopeDefining bool) err
...
 	// If explicit --type given, go straight to single-artifact install
 	if itemType != "" {
+		// run() already refuses --frozen --type as a usage error, so this is
+		// unreachable from the CLI. It is here so that refusal is a
+		// convenience rather than the only thing standing between --frozen and
+		// an install with no precheck, no pin match and no completeness check.
+		if installFrozen {
+			return frozenf("%s installs the agentpakke %s names, and %s takes one item regardless of what it says",
+				bold("--frozen"), agentpakke.DeclarationPath, bold("--type "+itemType))
+		}
```
[the remark this answers is not in the document]
Comment (Starefossen) [reply cli/nav-pilot/internal/cli/install.go:199]: Korrekt, og rettet i `95ac72b5`. Vakten under flaggnivået returnerer nå en vanlig `fmt.Errorf`, altså exit 1, slik flaggsjekken gjør.

Skillet er verdt å holde skarpt: motsigende flagg er en bruksfeil man retter på kommandolinja, mens `3` betyr at den erklærte pinnen ikke ble honorert. Det er to ulike ting for en CI-jobb å forgreine på, og én feil skal ikke ha to svar avhengig av hvilken dør den kom gjennom — samme grunn som `--user` ble samlet på 1 i forrige runde.

`TestFrozenRefusesTypeUnderneathTheFlagLayer` sjekker nå `ExitError`, og begge stedene i `docs/README.agentpakke.md` som beskriver koden er oppdatert. Vakten står, den er bare ikke lenger en annen kode: uten den ville `--type` gått rett til `cmdAdd` uten prekontroll, uten pinne-sammenlikning og uten fullstendighetssjekk.
